File: app/src/main/java/com/example/asynctask/ViewAdapter.java
package com.example.asynctask;

import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.bumptech.glide.Glide;
import com.example.asynctask.databinding.GridViewItemBinding;

import java.util.ArrayList;

public class ViewAdapter extends RecyclerView.Adapter<ViewAdapter.ItemViewHolder> {

    private ArrayList<ResultImage> images;

    public static class ItemViewHolder extends RecyclerView.ViewHolder {
        private GridViewItemBinding binding;

        public ItemViewHolder(@NonNull GridViewItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

        public void bind(ResultImage image) {
            Glide.with(binding.getRoot().getContext())
                    .load(image.getPreviewURL())
                    .centerCrop()
                    .into(binding.image);
        }
    }

    public void setData(ArrayList<ResultImage> image) {
        this.images = image;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public ItemViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        GridViewItemBinding binding = GridViewItemBinding.inflate(
                LayoutInflater.from(parent.getContext()), parent, false
        );
        return new ItemViewHolder(binding);
    }


    @Override
    public void onBindViewHolder(@NonNull ItemViewHolder holder, int position) {
        holder.bind(images.get(position));
    }

    @Override
    public int getItemCount() {
        return images.size();
    }
}

File: app/src/main/java/com/example/coroutine/ViewAdapter.kt
package com.example.coroutine

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.coroutine.databinding.GridViewItemBinding

class ViewAdapter(
    private var images: MutableList<ResultImage>
) : RecyclerView.Adapter<ViewAdapter.ItemViewHolder>() {

    inner class ItemViewHolder(private val binding: GridViewItemBinding) :
        RecyclerView.ViewHolder(binding.root) {

        fun bind(image: ResultImage) {
            Glide.with(binding.root.context)
                .load(image.previewURL)
                .centerCrop()
                .into(binding.image)
        }
    }

    private fun setData(image: MutableList<ResultImage>) {
        this.images = image
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ItemViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.grid_view_item, parent, false)
        return ItemViewHolder(GridViewItemBinding.bind(view))
    }

    override fun getItemCount(): Int {
        return images.size
    }

    override fun onBindViewHolder(holder: ItemViewHolder, position: Int) {
        holder.bind(images[position])
    }
}